test/test_icalevents.py

```diff
@@ -260,3 +286,62 @@ def test_event_categories(self):
         events = icalevents.events(url=None, file=ical, start=start, end=end)
         self.assertEqual(events[0].categories, ["In19-S04-IT2403"], "event 1 is not equal")
         self.assertEqual(events[1].categories, ["In19-S04-IT2406", "In19-S04-IT2405"], "event 2 is not equal")
+
+    def test_google_timezone(self):
+        ical = "test/test_data/google_tz.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        evs = icalevents.events(file=ical, start=start, end=end)
+
+        e1 = evs[0]
+        self.assertEqual(e1.start.hour, 0, "check start of the day")
+        self.assertEqual(e1.start.tzinfo, gettz('Europe/Zurich'), "check tz as specified in calendar")
+
+    def test_ms_timezone(self):
+        ical = "test/test_data/ms_tz.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        evs = icalevents.events(file=ical, start=start, end=end)
+
+        e1 = evs[0]
+        self.assertEqual(e1.start.hour, 0, "check start of the day")
+        self.assertEqual(e1.start.tzinfo, gettz('Europe/Berlin'), "check tz as specified in calendar")
+
+    def test_recurence_id_ms(self):
+        ical = "test/test_data/recurrenceid_ms.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        evs = icalevents.events(file=ical, start=start, end=end)
+
+        self.assertEqual(len(evs), 41, "41 events in total - one was moved")
+
+    def test_recurence_id_google(self):
+        ical = "test/test_data/recurrenceid_google.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        evs = icalevents.events(file=ical, start=start, end=end)
+
+        self.assertEqual(len(evs), 4, "4 events in total")
+
+    def test_cest(self):
+        ical = "test/test_data/cest.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        evs = icalevents.events(file=ical, start=start, end=end)
+
+        self.assertEqual(len(evs), 115, "4 events in total")
+
+    def test_transparent(self):
+        ical = "test/test_data/transparent.ics"
+        start = date(2021, 1, 1)
+        end = date(2021, 12, 31)
+
+        [e1, e2] = icalevents.events(file=ical, start=start, end=end)
+
+        self.assertEqual(e1.transparent, True, "respect transparency")
+        self.assertEqual(e2.transparent, False, "respect opaqueness")
```

Comment:
Nice work with adding quite a few extra tests. 
Considering the complexity of the parser I think we really need solid testing for everything we change :)